### PR TITLE
update(org): added focal-fossa support

### LINKED ISSUE #1823: Ubuntu 20.04 not supported by fo-installdeps script
### Description

Hi,

as requested in the "Install from sources" wiki, I'm reporting here that fo-installdeps script does not support my Ubuntu distro. (which is 20.04).

https://github.com/fossology/fossology/wiki/Install-from-Source

I guess there's no other information needed here :-)

#### How to reproduce

Run fo-installdeps on a Ubuntu 20.04 distro.

### Versions

[BUILD]
VERSION="3.9.0-rc2-2-g710b200f5"
BRANCH="master"
COMMIT_HASH=710b20
BUILD_DATE=2020/10/22 09:02 CEST
COMMIT_DATE=2020/10/14 09:07 CEST

No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal

### Logs
```
*** Installing both runtime and buildtime dependencies ***
install core dependencies
*** Installing Ubuntu buildtime dependencies ***
DB: Installing build essential.....
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'libglib2.0-cil-dev' for regex 'libglib2.0'
Note, selecting 'libglib2.0-tests' for regex 'libglib2.0'
Note, selecting 'libglib2.0-0-dbg' for regex 'libglib2.0'
Note, selecting 'libglib2.0-dev-bin' for regex 'libglib2.0'
Note, selecting 'libglib2.0-bin' for regex 'libglib2.0'
Note, selecting 'libglib2.0-cil' for regex 'libglib2.0'
Note, selecting 'libglib2.0-dev' for regex 'libglib2.0'
Note, selecting 'libglib2.0-doc' for regex 'libglib2.0'
Note, selecting 'libglib2.0-data' for regex 'libglib2.0'
Note, selecting 'libglib2.0-0' for regex 'libglib2.0'
build-essential is already the newest version (12.8ubuntu1).
libboost-program-options-dev is already the newest version (1.71.0.0ubuntu2).
libicu-dev is already the newest version (66.1-2ubuntu2).
libmagic-dev is already the newest version (1:5.38-4).
libxml2-dev is already the newest version (2.9.10+dfsg-5).
libboost-regex-dev is already the newest version (1.71.0.0ubuntu2).
libcunit1-dev is already the newest version (2.1-3-dfsg-2build1).
libglib2.0-cil is already the newest version (2.12.40-3).
libglib2.0-cil-dev is already the newest version (2.12.40-3).
libmxml-dev is already the newest version (3.1-1).
librpm-dev is already the newest version (4.14.2.1+dfsg1-1build2).
libtext-template-perl is already the newest version (1.58-1).
rpm is already the newest version (4.14.2.1+dfsg1-1build2).
subversion is already the newest version (1.13.0-3).
curl is already the newest version (7.68.0-1ubuntu2.2).
libglib2.0-0 is already the newest version (2.64.3-1~ubuntu20.04.1).
libglib2.0-bin is already the newest version (2.64.3-1~ubuntu20.04.1).
libglib2.0-data is already the newest version (2.64.3-1~ubuntu20.04.1).
libglib2.0-dev is already the newest version (2.64.3-1~ubuntu20.04.1).
libglib2.0-dev-bin is already the newest version (2.64.3-1~ubuntu20.04.1).
libglib2.0-doc is already the newest version (2.64.3-1~ubuntu20.04.1).
libpq-dev is already the newest version (12.4-0ubuntu0.20.04.1).
libglib2.0-tests is already the newest version (2.64.3-1~ubuntu20.04.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
ERROR: Unknown or Unsupported Ubuntu focal release, please report to the mailing list
```

#### FOSSology logs

Logs generated under /var/log/fossology/fossology.log

#### Apache logs

Logs generated under /var/log/apache2/error.log

#### Job logs

Logs generated under Geeky Scan Details (Job history => click on agent's job id)

### PR DESCRIPTION
**Pull request by sjha2048**
## Description
Added Ubuntu 20.04 support 

Fixes #1823
Part of #1760
Previous Conversation: #1765

### CODE REVIEW

**Comment by GMishx** [comment]
Now since `jessie` is gone, every other Ubuntu/Debian distro supports `composer`. So shall we add it to the this list?
https://github.com/fossology/fossology/blob/e18c3e99845064d21553c4a83cfd2f405aed574f/utils/fo-installdeps#L101-L107

cc @ag4ums

File: utils/fo-installdeps [hunk 1]
```diff
@@ -172,23 +176,26 @@ if [[ $RUNTIME ]]; then
...
                apt-get $YesOpt install postgresql-10 php7.2 php7.2-pgsql libapache2-mod-php7.2 php7.2-pgsql \
-                 php7.2-cli php7.2-curl php7.2-xml php7.2-zip php7.2-mbstring s-nail libboost-program-options1.65.1 \
+                 php7.2-cli php7.2-curl php7.2-xml php7.2-zip php7.2-mbstring php-gettext s-nail libboost-program-options1.65.1 \
                  libboost-regex1.65.1 libicu60;;
             sid)
-               apt-get $YesOpt install postgresql php php-pgsql php-cli php-curl php-xml php-zip php-mbstring s-nail \
+               apt-get $YesOpt install postgresql php php-pgsql php-cli php-curl php-xml php-zip php-mbstring php-gettext s-nail \
                  libboost-program-options1.67.0 libboost-regex1.67.0 libicu63;;
+            focal)
+               apt-get $YesOpt install postgresql php php-pgsql php-cli php-curl php-xml php-zip php-mbstring gettext s-nail \
```
Comment (GMishx) [inline utils/fo-installdeps:197]: The `gettext` package is no longer required here. Just tested on Ubuntu Focal and everything works without it.

File: src/nomos/mod_deps [hunk 2]
```diff
@@ -96,6 +96,8 @@ if [[ $RUNTIME ]]; then
           apt-get $YesOpt install libjson-c2;;
         stretch|buster|sid|bionic|cosmic)
           apt-get $YesOpt install libjson-c3;;
+        focal)
+           apt-get $YesOpt install libjson-c4;;
```
Comment (GMishx) [inline src/nomos/mod_deps:100]: Small indentation error.

```suggestion
          apt-get $YesOpt install libjson-c4;;
```

**Review by GMishx — CHANGES_REQUESTED** [review]
Looks like `debian/control` file will require changes too to build Debian packages.

Please check the `Build-Depends` and `Depends` tags for the list of packages to be updated.

File: utils/fo-installdeps (hunk 1 above)
Comment (sjha2048) [reply utils/fo-installdeps:197]: Okay, I'll remove it

**Comment by sjha2048** [comment]
> Now since `jessie` is gone, every other Ubuntu/Debian distro supports `composer`. So shall we add it to the this list?
> https://github.com/fossology/fossology/blob/e18c3e99845064d21553c4a83cfd2f405aed574f/utils/fo-installdeps#L101-L107
> 
> cc @ag4ums

Shall I add it then?

**Comment by sjha2048** [comment]
@GMishx please check the control file once, I'm not so sure about it.

**Comment by GMishx** [comment]
Looks perfect. Just 3 small changes:
1. Please change the second commit message to comply with contributions guidelines.
1. Please add `composer` to `Build-Depends` in control file.
1. Please remove `php5` from the control file as well as only jessie supported it and now its removed.

**Comment by sjha2048** [comment]
Updated and squashed all the commits, please check it once now.

**Review by GMishx — APPROVED** [review]
Changes looks good.

Tested building packages on Ubuntu Xenial, Bionic, Focal and Debian Buster, Stretch.
Also tested running `utils/fo-installdeps`

**Review by ag4ums — APPROVED** [review]
changes looks good, 
installed on ubuntu:20.04 and work as expected.

**Pull request merged**